File: welvet/python/examples/run_all.py
#!/usr/bin/env python3
"""Run every script in examples/ (used by README and CI smoke)."""

import subprocess
import sys
from pathlib import Path

EXAMPLES = [
    "01_dense_forward.py",
    "02_morph_and_train.py",
    "03_save_reload.py",
    "04_mha_forward.py",
    "05_dna_compare.py",
]


def main() -> int:
    root = Path(__file__).resolve().parent
    py = sys.executable
    failed = []

    print("welvet examples — running %d scripts\n" % len(EXAMPLES))
    for name in EXAMPLES:
        path = root / name
        print("→ %s" % name)
        r = subprocess.run([py, str(path)], cwd=root.parent)
        if r.returncode != 0:
            failed.append(name)
            print("  ❌ failed (exit %d)\n" % r.returncode)
        else:
            print("  ✓\n")

    if failed:
        print("FAILED: %s" % ", ".join(failed))
        return 1
    print("✅ all examples passed")
    return 0


if __name__ == "__main__":
    sys.exit(main())

File: welvet/dart/examples/run_all.dart
// Run all README examples (mirrors python/examples/run_all.py).
//
//   cd examples && dart run run_all.dart

import 'dart:io';

final scripts = [
  '01_dense_forward.dart',
  '02_morph_and_train.dart',
  '03_save_reload.dart',
  '04_mha_forward.dart',
  '05_dna_compare.dart',
];

Future<void> main() async {
  final dir = Directory.current.path;
  var failed = 0;
  for (final script in scripts) {
    print('── $script');
    final result = await Process.run('dart', ['run', script], workingDirectory: dir);
    stdout.write(result.stdout);
    stderr.write(result.stderr);
    if (result.exitCode != 0) {
      print('FAILED: $script (exit ${result.exitCode})');
      failed++;
    }
  }
  if (failed > 0) {
    print('\n❌ $failed example(s) failed');
    exit(1);
  }
  print('\n✅ all examples passed');
}
